File: client-examples/python/auth_console_app.py
#!/usr/bin/env python3
"""
Null-Auth Console Application Sample (Python) with Version Checker
Demonstrates a clean Console Application implementing BOTH Auth Methods + Version Checking:
  - Method 1: License Key Authentication + Version Validation
  - Method 2: Direct HWID Whitelist Authentication + Version Validation
"""

import sys
import json
from null_auth_sdk import NullAuthSDK

# Configure App Credentials & Current Client Version
APP_ID = "NA-48392017"                  # Replace with your App ID
APP_SECRET = "nas_YOUR_APP_SECRET_HERE" # Replace with your App Secret
API_URL = "https://null-auth-backend.vercel.app"
APP_VERSION = "1.0.0"                   # Current local application version

def main():
    print("=================================================")
    print("    🛡️ Null-Auth Console App (Version " + APP_VERSION + ")     ")
    print("=================================================")

    sdk = NullAuthSDK(APP_ID, APP_SECRET, API_URL)
    user_sid = sdk.get_windows_user_sid()

    print(f"\n[+] Detected Windows User SID: {user_sid}")
    print(f"[+] Client Version: {APP_VERSION}")

    print("\nSelect Authentication Method:")
    print("  1. Method 1: License Key + Bound Machine SID")
    print("  2. Method 2: HWID Whitelist Only (No License Key)")

    choice = input("\nEnter Choice (1 or 2): ").strip()

    if choice == "1":
        license_key = input("\nEnter License Key (e.g. NULL-ABCD-1234-EFGH): ").strip()
        print("\n[*] Authenticating License Key with Null-Auth Cloud Server...")
        result = sdk.authenticate_license(license_key, client_version=APP_VERSION)
    else:
        print("\n[*] Authenticating HWID Whitelist with Null-Auth Cloud Server...")
        result = sdk.authenticate_hwid(client_version=APP_VERSION)

    print("\n-------------------------------------------------")

    # Check for Version Mismatch
    if result.get("error") == "VERSION_MISMATCH":
        required_ver = result.get("data", {}).get("requiredVersion", "Unknown")
        download_url = result.get("data", {}).get("downloadUrl")
        print("\n[!] CRITICAL: VERSION MISMATCH DETECTED!")
        print(f"    Your App Version: {APP_VERSION}")
        print(f"    Required Version: {required_ver}")
        if download_url:
            print(f"    Download Update: {download_url}")
        print("\n[-] EXECUTION TERMINATED: Application is outdated.")
        sys.exit(1)

    if result.get("success"):
        data = result.get("data", {})
        print("\n[+] ACCESS GRANTED! Application Successfully Unlocked.")
        print(f"    Status: {data.get('status')}")
        print(f"    Expires At: {data.get('expires_at')}")
        print(f"    Remaining: {data.get('remaining_days')} Days")
    else:
        reason = result.get("message", "Authentication Failed")
        print(f"\n[-] ACCESS DENIED! Reason: {reason}")
        sys.exit(1)

    print("-------------------------------------------------")
    print("\n[+] Starting Main Software Application Payload...")
    input("\nPress Enter to exit...")

if __name__ == "__main__":
    main()

File: client-examples/cpp/AuthConsoleApp.cpp
#include <iostream>
#include <string>
#include "NullAuthSDK.hpp"

/**
 * C++ Console Application Sample with Version Checker:
 *   - Method 1: License Key Authentication + Version Validation
 *   - Method 2: Direct HWID Whitelist Authentication + Version Validation
 */

int main() {
    SetConsoleTitleA("Null-Auth C++ Console Client");

    std::string version = "1.0.0"; // Current local application version

    std::cout << "=================================================\n";
    std::cout << "  🛡️ Null-Auth C++ Console Client (v" << version << ")   \n";
    std::cout << "=================================================\n\n";

    std::string host = "null-auth-backend.vercel.app";
    std::string appId = "NA-48392017";                  // Replace with your App ID
    std::string appSecret = "nas_YOUR_APP_SECRET_HERE"; // Replace with your App Secret

    std::string userSid = NullAuth::GetWindowsUserSid();
    std::cout << "[+] Detected Windows User SID: " << userSid << "\n";
    std::cout << "[+] Client Version: " << version << "\n\n";

    std::cout << "Select Authentication Method:\n";
    std::cout << "  1. Method 1: License Key + Bound Machine SID\n";
    std::cout << "  2. Method 2: HWID Whitelist Only (No License Key)\n";
    std::cout << "\nEnter Choice (1 or 2): ";
    
    int choice = 1;
    std::cin >> choice;

    std::string response;
    bool sent = false;

    if (choice == 1) {
        std::cout << "\nEnter License Key (e.g. NULL-ABCD-1234-EFGH): ";
        std::string licenseKey;
        std::cin >> licenseKey;

        std::cout << "\n[*] Authenticating License Key with Null-Auth Cloud Server...\n";
        sent = NullAuth::AuthenticateLicense(host, appId, appSecret, licenseKey, version, response);
    } else {
        std::cout << "\n[*] Authenticating HWID Whitelist with Null-Auth Cloud Server...\n";
        sent = NullAuth::AuthenticateHwid(host, appId, appSecret, version, response);
    }

    std::cout << "\n-------------------------------------------------\n";
    std::cout << "[Server Raw Response]: " << response << "\n";

    // Version Mismatch Check
    if (response.find("VERSION_MISMATCH") != std::string::npos) {
        std::cout << "\n[!] CRITICAL: VERSION MISMATCH DETECTED!\n";
        std::cout << "[-] EXECUTION TERMINATED: Application is outdated.\n";
        std::cout << "-------------------------------------------------\n";
        std::cout << "\nPress Enter to exit...";
        std::cin.ignore();
        std::cin.get();
        return 1;
    }

    if (sent && response.find("\"success\":true") != std::string::npos) {
        std::cout << "\n[+] ACCESS GRANTED! Application Successfully Unlocked.\n";
        std::cout << "[+] Starting Main Software Application Payload...\n";
    } else {
        std::cout << "\n[-] ACCESS DENIED! Invalid Credentials or Unauthorized Machine.\n";
    }

    std::cout << "-------------------------------------------------\n";
    std::cout << "\nPress Enter to exit...";
    std::cin.ignore();
    std::cin.get();
    return 0;
}
